homework_7.py

```diff
@@ -0,0 +1,120 @@
+""""HOMEWORK #7. Andrei Kulbitski"""
+
+import random
+
+# 1. Быки и коровы¶
+print("TASK 1: BULLS AND COWS")
+
+
+def magic_number_generator_func():
+    """Random unique 4-digit generator"""
+    digits = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]
+    random.shuffle(digits)
+    return digits[:4]
+
+
+# Magic number is stored as four integers packaged in the list
+magic_number = magic_number_generator_func()
+
+# User is expected to guess the number in four attempts :)
+attempts = 8
+
+while True:
+    # Position counter for simultaneous enumeration of two lists
+    position_key = 0
+    # Bulls - amount of full alignments (digit and position)
+    count_of_bulls = 0
+    # Cows - amount of partial alignments (digit only)
+    count_of_cows = 0
+
+    # Handling user input:
+    # Every non-4-digit input will be rejected
+    # and user would be forced to retry
+    current_attempt = input(
+        "Try to guess the magical number. Enter four digits: ... ")
+    current_attempt_list = []
+
+    if not current_attempt.isdigit() or not len(current_attempt) == 4:
+        print("Wrong input: 4 digits expected. "
+              "Please, follow the next pattern: XXXX.")
+        continue
+
+    for i in current_attempt:
+        # User input is stored as four integers packaged in the list
+        current_attempt_list.append(int(i))
+
+    for digit_b in magic_number:
+        # Loop for calculation of amount of bulls
+        if digit_b == current_attempt_list[position_key]:
+            count_of_bulls += 1
+        position_key += 1
+    print("Bulls:", count_of_bulls)
+
+    for digit_c in magic_number:
```

Comment:
cows calculation does not work correctly

Reply:
Fixed it, now: count_of_cows = cows - bulls